### ISSUE TITLE
Support for lapack 3.10.0

### ISSUE THREAD

**Issue by h-vetinari** [conda-forge/intel_repack-feedstock#26]
Following up from #6 / #24, opening an issue for tracking MKL compatibility with the recently [released](https://github.com/Reference-LAPACK/lapack/releases/tag/v3.10.0) lapack 3.10.0.

CC @sknepper @PetrovKP

Xref:
https://github.com/conda-forge/lapack-feedstock/pull/55
https://github.com/conda-forge/blas-feedstock/pull/72

**Comment by h-vetinari** [comment]
The statement from @sknepper in #32 was:
> Intel oneMKL 2022.0 does not support LAPACK 3.9.1/3.10. The soonest that would be supported would be in Intel oneMKL 2022.1. The oneMKL release notes will include this change once it is made.

Now I see the following in the 2022.1 [release notes](https://www.intel.com/content/www/us/en/developer/articles/release-notes/onemkl-release-notes.html):

> * LAPACK
>   * [...]
>   * Introduced multishift QZ algorithm from Netlib LAPACK 3.10.0; Integrated other minor bug fixes from Netlib LAPACK 3.9.1 – 3.10.
>   * Modified a set of ?LAQR[0-5] computational kernels used for solving non-symmetric eigenvalue problems (?GEEV, ?GEES, ?GEEVX and ?GEESX) as was done in Netlib LAPACK 3.10. 

Does this imply full Lapack 3.10 compatibility, or is anything still missing? It sounds like everything is ready, but just to double-check...

**Comment by sknepper** [comment]
Hi @h-vetinari,
Most of the changes from Netlib LAPACK 3.10.0 are in Intel oneMKL 2022.1; however, the new Householder reconstruction-related functionality (GETSQRHRT and related) is not available, so we are not fully aligned.

**Comment by h-vetinari** [comment]
Also, have you looked at Lapack [3.10.1](http://netlib.org/lapack/lapack-3.10.1.html) already? It advertises itself as:
> This is a minor release addressing multiple bug fixes and improvements.

**Comment by sknepper** [comment]
Hi @h-vetinari ,
It's not quite correct to say that Intel oneMKL 2022.1 has full compatibility with Lapack 3.9.1. There was some new functionality introduced in 3.9.1 that is not yet available in Intel oneMKL.
Based on the changes in Lapack 3.10.1, we expect to include these in the same release as the remaining functionality from Lapack 3.9.1/3.10.0, which would then bring alignment/full compatibility with Lapack 3.10.1. Thanks!

**Comment by sknepper** [comment]
Hello @h-vetinari ,
The newly-released Intel oneMKL 2022.2 is aligned with Netlib LAPACK 3.10.1, currently the latest release of Netlib LAPACK.
(If interested in ScaLAPACK, Intel oneMKL 2022.2 is aligned with Netlib ScaLAPACK 2.2.0, the latest release of Netlib ScaLAPACK.) Thank you! =)

**Comment by alippai** [comment]
@h-vetinari @sknepper I assume LAPACK 3.10.1 is ready, LAPACK 3.11 still needs work, right?

**Comment by sknepper** [comment]
Hi @alippai - yes, you are correct. Are there particular new features or fixes from LAPACK 3.11 that you are interested in?

**Comment by alippai** [comment]
Not necessarily. I've found this issue when creating a conda environment and saw the weird version number 3.9. Checked the upstream versions and wanted a confirmation what's coming next (based on your answer it's 3.10.1).

**Comment by sknepper** [comment]
Thanks for the clarification, @alippai . Intel oneMKL 2022.2 and Intel oneMKL 2023.0 (currently the latest release of Intel oneMKL) are both aligned to Netlib LAPACK 3.10.1, and calling [ilaver](https://www.intel.com/content/www/us/en/docs/onemkl/developer-reference-fortran/2023-0/ilaver.html) should reflect that.

**Comment by h-vetinari** [comment]
> Intel oneMKL 2022.2 and Intel oneMKL 2023.0 (currently the latest release of Intel oneMKL) are both aligned to Netlib LAPACK 3.10.1

Thanks for the update @sknepper, sorry it took me a while to get around to it. Unfortunately there are some test failures when running MKL 2023.1 against the LAPACK 3.10.1 test suite: https://github.com/conda-forge/intel_repack-feedstock/issues/44 - your help would be much appreciated!

BTW, I renamed this issue to be a tracking issue for LAPACK support in MKL, as this will be an ongoing concern (i.e. LAPACK 3.11.0 has been out for a while, and we similarly would want to know when MKL will support that).

**Comment by h-vetinari** [comment]
@conda-forge/intel_repack
Given the persistent and blocking problems with running the LAPACK 3.10 test suite (#44), I'm wondering what the status about compatibility with LAPACK 3.11 resp. 3.12 is? Is there an ETA for compatibility with those? For example, if the errors are gone with 3.11, it might make sense to jump to that directly

**Comment by h-vetinari** [comment]
> Given the persistent and blocking problems with running the LAPACK 3.10 test suite (#44), I'm wondering what the status about compatibility with LAPACK 3.11 resp. 3.12 is? Is there an ETA for compatibility with those?

Please provide an update @sknepper @napetrov @oleksandr-pavlyk

**Comment by sknepper** [comment]
Hi @h-vetinari - the upcoming Intel oneMKL 2024.1 release will be aligned to Netlib LAPACK 3.11. We don't currently have an ETA for compatibility with Netlib LAPACK 3.12.

**Comment by h-vetinari** [comment]
Thank you very much for the update!

That means we might skip LAPACK 3.10 directly, assuming the test failures seen in #44 for MKL + LAPACK 3.10 would go away with LAPACK 3.11. Still, I'd appreciate it a lot if you could have a look at the failures there in any case, as this has been blocking our infrastructure from upgrading LAPACK for almost a year now (in addition to waiting for MKL compatibility with 3.10 in the first place).
